src/entities/postgres-storage/database.js

```diff
@@ -480,24 +497,39 @@ class DB implements Database {
     this.#logger.debug(`storeBlockTxs (${epoch}/${String(slot)}, ${hash}, ${block.getHeight()})`)
     const newUtxos = utils.getTxsUtxos(txs)
     const blockUtxos = []
-    // TODO: implement for accounts
-    const requiredInputs = _.flatMap(txs, tx => tx.inputs).filter(inp => {
-      const utxoId = utils.getUtxoId(inp)
-      const localUtxo = newUtxos[utxoId]
-      if (localUtxo) {
-        blockUtxos.push({
-          id: localUtxo.utxo_id,
-          address: localUtxo.receiver,
-          amount: localUtxo.amount,
-          txHash: localUtxo.tx_hash,
-          index: localUtxo.tx_index,
-        })
-        // Delete new Utxo if it's already spent in the same block
-        delete newUtxos[utxoId]
-        // Remove this input from required
+    // TODO: these are some changes I made quickly to see if I could protoype account spending
+    // and the code is not exactly of the best quality.
+    // indexed by tx index in block
+    const accountInputs: Map<number, Array<AccountType>> = new Map()
+    const withdrawls: Map<string, Number> = new Map()
+    const requiredInputs = _.flatMap(txs, (tx) => tx.inputs).filter((inp, txOrdinal) => {
+      if (inp.type === 'utxo') {
+        const utxoId = utils.getUtxoId(inp)
+        const localUtxo = newUtxos[utxoId]
+        if (localUtxo) {
+          blockUtxos.push({
+            id: localUtxo.utxo_id,
+            address: localUtxo.receiver,
+            amount: localUtxo.amount,
+            txHash: localUtxo.tx_hash,
+            index: localUtxo.tx_index,
+          })
+          // Delete new Utxo if it's already spent in the same block
+          delete newUtxos[utxoId]
+          // Remove this input from required
+          return false
+        }
+        return true
+      }
+      if (inp.type === 'account') {
+        if (typeof accountInputs.get(txOrdinal) === 'undefined') {
+          accountInputs.set(txOrdinal, [])
+        }
+        accountInputs.get(txOrdinal).push(inp)
+        withdrawls.set(inp.account_id, (withdrawls.get(inp.account_id) || 0) + inp.value)
```

Comment:
This part here will not work as intended right now, cuz we are doing `flatMap` on all inputs, so `txOrdinal` will be unique per each input, instead of per tx.

Reply:
Discussed with @crchemist-ip - this part must be reworked in the future about adding account support